### PR TITLE
LOG-5276: Enable functional json parse test as filter

### PR DESCRIPTION
**Pull request by jcantrill**
### Description
This PR:
* Enables the JSON functional test
* Moves JSON parsing into a filter package
* Removes obsolete functional tests
* Enables JSON parsing for any container log
* Enables some filter unit tests in the functional directory

### Links
* https://issues.redhat.com/browse/LOG-5726
* blocked by #2591

### CODE REVIEW

**Comment by jcantrill** [comment]
cc @Clee2691 @cahartma

File: test/functional/filters/parse/suite_test.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package parse
+
+import (
+	"testing"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+func TestNormalization(t *testing.T) {
```
Comment (Clee2691) [inline test/functional/filters/parse/suite_test.go:10]: Nit:
```suggestion
func TestParse(t *testing.T) {
```

**Pull request merged**